### PR TITLE
libgc: Update to 8.2.2

### PR DESCRIPTION
**Pull request by ivmai**
* Fix license (the package is covered by MIT-style license, not GPL)
* Drop include-private-gcconfig.h.patch (should not be needed any more)
* Remove dependency on libatomic_ops (libgc uses gcc built-in atomics)

### CODE REVIEW

**Comment by ravener** [comment]
> Remove dependency on libatomic_ops (libgc uses gcc built-in atomics)

Termux does not use `gcc` though, it uses `clang` from the Android NDK.

**Comment by ivmai** [comment]
> Termux does not use `gcc` though, it uses `clang` from the Android NDK.

clang is compliant with gcc regarding atomic intrinsic.

**Comment by ravener** [comment]
Well somehow the CI logs says that the build failed because libatomic_ops is required.

**Comment by ivmai** [comment]
> checking for compiler intrinsics support... skipped because cross-compiling

I'll fix it by passing --with-libatomic-ops=none to configure

**Comment by ivmai** [comment]
@ravener How to pass an extra flag to configure correctly?

**Comment by 2096779623** [comment]
> @ravener How to pass an extra flag to configure correctly?

> If you need to pass some additional arguments, use the field `TERMUX_PKG_EXTRA_CONFIGURE_ARGS` .

https://github.com/termux/termux-packages/wiki/Creating-new-package

**Comment by 2096779623** [comment]
Try add `TERMUX_PKG_LICENSE_FILE="../../../LICENSE"` to install LICENSE file.

**Comment by ivmai** [comment]
> Try add `TERMUX_PKG_LICENSE_FILE="../../../LICENSE"` to install LICENSE file.
> 
> Edit: Maybe `TERMUX_PKG_BUILD_IN_SRC=true` is more suitable.

I've decided finally to install README.QUICK which lists the license.

File: packages/libgc/build.sh [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=(8.0.6
-                    7.6.12)
-TERMUX_PKG_SRCURL=(https://github.com/ivmai/bdwgc/releases/download/v$TERMUX_PKG_VERSION/gc-$TERMUX_PKG_VERSION.tar.gz
-                   https://github.com/ivmai/libatomic_ops/releases/download/v${TERMUX_PKG_VERSION[1]}/libatomic_ops-${TERMUX_PKG_VERSION[1]}.tar.gz)
-TERMUX_PKG_SHA256=(3b4914abc9fa76593596773e4da671d7ed4d5390e3d46fbf2e5f155e121bea11
-                   f0ab566e25fce08b560e1feab6a3db01db4a38e5bc687804334ef3920c549f3e)
+TERMUX_PKG_VERSION=(8.2.2)
+TERMUX_PKG_SRCURL=(https://github.com/ivmai/bdwgc/releases/download/v$TERMUX_PKG_VERSION/gc-$TERMUX_PKG_VERSION.tar.gz)
+TERMUX_PKG_SHA256=(f30107bcb062e0920a790ffffa56d9512348546859364c23a14be264b38836a0)
```
[the remark this answers is not in the document]
Comment (ivmai) [reply packages/libgc/build.sh]: Done

**Review by buttaface — APPROVED** [review]

**Comment by ivmai** [comment]
@ravener, @2096779623 Anything else to change?

**Pull request merged**